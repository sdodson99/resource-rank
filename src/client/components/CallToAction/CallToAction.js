import React from 'react';
import styles from './CallToAction.module.css';
import Link from 'next/link';

const CallToAction = () => (
  <div className={styles.CallToAction} data-testid="CallToAction">
    <div className={`content-container ${styles.Content}`}>
      <div className="text-6xl">What are you waiting for?</div>
      <div className="mt-10">
        <Link href="/topics">
          <a className="btn btn-primary">Explore Topics</a>
        </Link>
      </div>
    </div>
    <div className={styles.ColorOverlay}></div>
  </div>
);

CallToAction.propTypes = {};

CallToAction.defaultProps = {};

export default CallToAction;
