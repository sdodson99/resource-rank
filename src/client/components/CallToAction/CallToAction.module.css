.CallToAction {
  @apply py-16 bg-gray-100 text-center;
}

.Search {
  max-width: 600px;

  @apply mt-10 mx-auto flex flex-col sm:flex-row;
}
