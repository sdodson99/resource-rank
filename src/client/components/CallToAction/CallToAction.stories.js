import React from 'react';
import CallToAction from './CallToAction';

export default {
  component: CallToAction,
  title: 'Components/CallToAction',
  decorators: [
    (Story) => {
      return <Story />;
    },
  ],
};

const Template = (args) => <CallToAction {...args} />;

export const Default = Template.bind({});
Default.args = {};
