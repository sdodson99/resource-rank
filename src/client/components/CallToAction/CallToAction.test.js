import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import CallToAction from './CallToAction';
import { useRouter } from 'next/router';

jest.mock('next/router');

describe('<CallToAction />', () => {
  let mockRouterPush;

  beforeEach(() => {
    mockRouterPush = jest.fn();

    useRouter.mockReturnValue({
      push: mockRouterPush,
    });
  });

  afterEach(() => {
    useRouter.mockReset();
  });

  test('it should mount', () => {
    render(<CallToAction />);

    const callToAction = screen.getByTestId('CallToAction');

    expect(callToAction).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const tree = renderer.create(<CallToAction />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should navigate to topic search when search button clicked', () => {
    const search = 'topic123';
    render(<CallToAction />);
    const searchInput = screen.getByPlaceholderText('Search...');
    const searchButton = screen.getByText('Search Topics');

    fireEvent.change(searchInput, {
      target: {
        value: search,
      },
    });
    searchButton.click();

    expect(mockRouterPush).toBeCalledWith(`/topics?q=${search}`);
  });
});
