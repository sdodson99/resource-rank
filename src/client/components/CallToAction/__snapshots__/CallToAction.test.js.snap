// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CallToAction /> should render correctly 1`] = `
<div
  className="CallToAction"
  data-testid="CallToAction"
>
  <div
    className="content-container"
  >
    <div
      className="text-6xl"
    >
      What are you waiting for?
    </div>
    <div
      className="Search"
    >
      <input
        className="flex-grow block form-control-lg"
        onChange={[Function]}
        placeholder="Search..."
        type="text"
        value=""
      />
      <button
        className="mt-5 sm:mt-0 sm:ml-3 btn btn-primary"
        onClick={[Function]}
      >
        Search Topics
      </button>
    </div>
  </div>
</div>
`;
